python/cudf/dataframe/dataframe.py

```diff
@@ -2466,8 +2483,9 @@ def __getitem__(self, arg):
 
         for col in self._df.columns:
             sr = self._df[col]
-            df.add_column(col, sr.iloc[tuple(rows)], forceindex=True)
+            df.add_column(col, sr.iloc[tuple(rows)])
 
+        df.index = sr.index.take(np.array(rows))
```

Comment:
@kkraus14 is there a better way of doing this ?

Reply:
Ideally we should implement Iloc for the Index class as well, but this is one of the places where a refactor is really needed. You should be able to get away with standard `__getitem__` semantics here I believe. It should pass the call down to the `Column` object.

Reply:
Do you mean something like:
> sr.index[np.array(rows)]  

?

Reply:
> Do you mean something like:
> 
> > sr.index[np.array(rows)]
> 
> ?

Yes, but in looking at this it may not be fully implemented / robust quite yet.